File: languages/php/Bloock/Entity/Key/ManagedKey.php
<?php

namespace Bloock\Entity\Key;

use Bloock\Entity\Key\AccessControlType;

/**
 * Represents a managed key.
 */
class ManagedKey
{
    /**
     * Is the unique identifier of the managed key (ex: 46c49ee7-ef44-472c-a873-ce81a2d5d764).
     * @var string
     */
    public string $id;
    /**
     * Is the name of the managed key.
     * @var string|null
     */
    public ?string $name;
    /**
     * Is the protection level for the key.
     * @var string
     */
    public string $protection;
    /**
     * Is the type of the key.
     * @var string
     */
    public string $keyType;
    /**
     * Is the timestamp indicating when the key expires.
     * @var int|null
     */
    public ?int $expiration;
    /**
     * Is the actual public key.
     * @var string
     */
    public string $key;
    /**
     * Is the access control type for the key.
     * @var string
     */
    public string $accessControlType;

    /**
     * Constructs a ManagedKey object with the specified parameters.
     * @param string $id
     * @param string|null $name
     * @param string $protection
     * @param string $keyType
     * @param int|null $expiration
     * @param string $key
     * @param string $accessControlType
     */
    public function __construct(string $id, ?string $name, string $protection, string $keyType, ?int $expiration, string $key, string $accessControlType)
    {
        $this->id = $id;
        $this->name = $name;
        $this->protection = $protection;
        $this->keyType = $keyType;
        $this->expiration = $expiration;
        $this->key = $key;
        $this->accessControlType = $accessControlType;
    }

    public static function fromProto(\Bloock\ManagedKey $res): ManagedKey
    {
        return new ManagedKey($res->getId(), $res->getName(), KeyProtectionLevel::fromProto($res->getProtection()), KeyType::fromProto($res->getKeyType()), $res->getExpiration(), $res->getKey(), AccessControlType::fromProto($res->getAccessControlType()));
    }

    public function toProto(): \Bloock\ManagedKey
    {
        $p = new \Bloock\ManagedKey();
        $p->setId($this->id);
        $p->setName($this->name);
        $p->setProtection(KeyProtectionLevel::toProto($this->protection));
        $p->setKeyType(KeyType::toProto($this->keyType));
        $p->setExpiration($this->expiration);
        $p->setKey($this->key);
        $p->setAccessControlType(AccessControlType::toProto($this->protection));
        return $p;
    }
}

File: languages/python/bloock/entity/key/managed_key.py
from __future__ import annotations

import bloock._bridge.proto.keys_entities_pb2 as proto
from bloock.entity.key.key_protection_level import KeyProtectionLevel
from bloock.entity.key.key_type import KeyType
from bloock.entity.key.access_control_type import AccessControlType


class ManagedKey:
    """
    Represents a managed key.
    """
    def __init__(
            self,
            id: str,
            name: str,
            protection: KeyProtectionLevel,
            key_type: KeyType,
            expiration: int,
            key: str,
            access_control_type: AccessControlType,
    ) -> None:
        """
        Constructs a ManagedKey object with the specified parameters.
        :type key: object
        :type key_type: object
        :type expiration: object
        :type protection: object
        :type access_control_type: object
        :type name: object
        :type id: object
        """
        self.id = id
        self.name = name
        self.protection = protection
        self.key_type = key_type
        self.expiration = expiration
        self.key = key
        self.access_control_type = access_control_type

    @staticmethod
    def from_proto(key: proto.ManagedKey) -> ManagedKey:
        return ManagedKey(
            id=key.id,
            name=key.name,
            protection=KeyProtectionLevel.from_proto(key.protection),
            key_type=KeyType.from_proto(key.key_type),
            expiration=key.expiration,
            key=key.key,
            access_control_type=AccessControlType.from_proto(key.access_control_type)
        )

    def to_proto(self) -> proto.ManagedKey:
        return proto.ManagedKey(
            id=self.id,
            name=self.name,
            protection=self.protection.to_proto(),
            key_type=self.key_type.to_proto(),
            expiration=self.expiration,
            key=self.key,
            access_control_type=self.access_control_type.to_proto(),
        )
